diff --git a/source/common/aarch64/asm.S b/source/common/aarch64/asm.S
index 399c37cf2..b81fb254e 100644
--- a/source/common/aarch64/asm.S
+++ b/source/common/aarch64/asm.S
@@ -28,6 +28,11 @@
 #define PFX2(prefix, name) PFX3(prefix, name)
 #define PFX(name)          PFX2(X265_NS, name)
 
+
+#define UPFX3(prefix, name) _ ## prefix ## _ ## name
+#define UPFX2(prefix, name) UPFX3(prefix, name)
+#define UPFX(name)          UPFX2(X265_NS, name)
+
 #ifdef __APPLE__
 #define PREFIX 1
 #endif
diff --git a/source/common/aarch64/pixel-util.S b/source/common/aarch64/pixel-util.S
index fba9a90d5..49c6b0492 100644
--- a/source/common/aarch64/pixel-util.S
+++ b/source/common/aarch64/pixel-util.S
@@ -2407,7 +2407,7 @@ function PFX(costCoeffNxN_neon)
     // x5 - scanFlagMask
     // x6 - baseCtx
     mov             x0, #0
-    movrel          x1, x265_entropyStateBits
+    movrel          x1, UPFX(entropyStateBits)
     mov             x4, #0
     mov             x11, #0
     movi            v31.16b, #0
