diff -ruN x265_git/source/common/version.cpp x265_git_quant/source/common/version.cpp
--- x265_git/source/common/version.cpp	2021-12-04 15:16:15.000000000 +0000
+++ x265_git_quant/source/common/version.cpp	2021-12-04 16:09:30.000000000 +0000
@@ -31,7 +31,7 @@
 
 #if defined(__clang__)
 #define COMPILEDBY  "[clang " XSTR(__clang_major__) "." XSTR(__clang_minor__) "." XSTR(__clang_patchlevel__) "]"
-#ifdef __IA64__
+#if defined(__IA64__) || defined(__arm64__) || defined(__aarch64__)
 #define ONARCH    "[on 64-bit] "
 #else
 #define ONARCH    "[on 32-bit] "
@@ -40,7 +40,7 @@
 
 #if defined(__GNUC__) && !defined(__INTEL_COMPILER) && !defined(__clang__)
 #define COMPILEDBY  "[GCC " XSTR(__GNUC__) "." XSTR(__GNUC_MINOR__) "." XSTR(__GNUC_PATCHLEVEL__) "]"
-#ifdef __IA64__
+#if defined(__IA64__) || defined(__arm64__) || defined(__aarch64__
 #define ONARCH    "[on 64-bit] "
 #else
 #define ONARCH    "[on 32-bit] "
@@ -71,7 +71,7 @@
 #define ONOS    "[Unk-OS]"
 #endif
 
-#if X86_64
+#if X86_64  || __arm64__ || __aarch64__
 #define BITS    "[64 bit]"
 #else
 #define BITS    "[32 bit]"
